### PR TITLE
Remove causeless direct byte buffers usages.

### PR DESCRIPTION
**Pull request by MishaDemianenko**
Replace usages of direct byte buffers with their on heap twin.

### CODE REVIEW

File: community/io/src/test/java/org/neo4j/io/pagecache/PageCacheTest.java [hunk 1]
```diff
@@ -4416,7 +4416,7 @@ public void copyToHeapByteBufferFromReadPageCursorMustCheckBounds() throws Excep
     public void copyToDirectByteBufferFromReadPageCursorMustCheckBounds() throws Exception
     {
         configureStandardPageCache();
-        ByteBuffer buffer = ByteBuffer.allocateDirect( filePageSize );
+        ByteBuffer buffer = ByteBuffer.allocate( filePageSize );
```
Comment (chrisvest) [inline community/io/src/test/java/org/neo4j/io/pagecache/PageCacheTest.java]: No, these three tests are explicitly about copying to/from _direct_ byte buffers.

File: enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/catchup/storecopy/FileSender.java [hunk 2]
```diff
@@ -46,7 +46,7 @@
     FileSender( StoreResource resource )
     {
         this.resource = resource;
-        this.byteBuffer = ByteBuffer.allocateDirect( MAX_SIZE );
+        this.byteBuffer = ByteBuffer.allocate( MAX_SIZE );
```
Comment (chrisvest) [inline enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/catchup/storecopy/FileSender.java]: Might this not cause extra copying of the buffer contents when transferring data between network and storage?

File: enterprise/com/src/main/java/org/neo4j/com/storecopy/StoreCopyServer.java [hunk 3]
```diff
@@ -168,7 +168,7 @@ public RequestContext flushStoresAndStreamStoreFiles( String triggerName, StoreW
             {
                 lastAppliedTransaction = checkPointer.lastCheckPointedTransactionId();
                 monitor.startStreamingStoreFiles( storeCopyIdentifier );
-                ByteBuffer temporaryBuffer = ByteBuffer.allocateDirect( (int) ByteUnit.mebiBytes( 1 ) );
+                ByteBuffer temporaryBuffer = ByteBuffer.allocate( (int) ByteUnit.mebiBytes( 1 ) );
```
Comment (chrisvest) [inline enterprise/com/src/main/java/org/neo4j/com/storecopy/StoreCopyServer.java]: Same deal as with `FileSender`.

File: community/io/src/test/java/org/neo4j/io/pagecache/PageCacheTest.java [hunk 4]
```diff
@@ -4446,7 +4446,7 @@ public void copyToHeapByteBufferFromWritePageCursorMustCheckBounds() throws Exce
     public void copyToDirectByteBufferFromWritePageCursorMustCheckBounds() throws Exception
     {
         configureStandardPageCache();
-        ByteBuffer buffer = ByteBuffer.allocateDirect( filePageSize );
+        ByteBuffer buffer = ByteBuffer.allocate( filePageSize );
```
Comment (chrisvest) [inline community/io/src/test/java/org/neo4j/io/pagecache/PageCacheTest.java]: Also here.

File: community/io/src/test/java/org/neo4j/io/pagecache/PageCacheTest.java [hunk 5]
```diff
@@ -4554,7 +4554,7 @@ public void copyToReadOnlyHeapByteBufferMustThrow() throws Exception
     public void copyToReadOnlyDirectByteBufferMustThrow() throws Exception
     {
         configureStandardPageCache();
-        ByteBuffer buf = ByteBuffer.allocateDirect( filePageSize ).asReadOnlyBuffer();
+        ByteBuffer buf = ByteBuffer.allocate( filePageSize ).asReadOnlyBuffer();
```
Comment (chrisvest) [inline community/io/src/test/java/org/neo4j/io/pagecache/PageCacheTest.java]: And here.

**Review by chrisvest — CHANGES_REQUESTED** [review]

**Comment by MishaDemianenko** [comment]
@chrisvest  Since i can't tell based on any measurements that its not the case (since we do not have any so far) i will agree that using direct byte buffers for file/network read/write should be faster and will revert changes in those two places. Let it be the fight for later when will have some time to write benchmarks for those 2 places.

**Review by chrisvest — APPROVED** [review]

**Pull request merged**